src/libcrun/container.c

```diff
@@ -2016,12 +2016,80 @@ cleanup_watch (libcrun_context_t *context, runtime_spec_schema_config_schema *de
 }
 
 static int
-open_seccomp_output (const char *id, int *fd, bool readonly, const char *state_root, libcrun_error_t *err)
+store_seccomp_cache (const char *state_root, const char *id, char checksum[65], libcrun_error_t *err)
+{
+  cleanup_free char *seccomp_cache_dir = NULL;
+  cleanup_free char *seccomp_cache = NULL;
+  cleanup_free char *src_path = NULL;
+  cleanup_free char *dir = NULL;
+  struct stat st;
+  int ret;
+
+  dir = libcrun_get_state_directory (state_root, id);
+  if (UNLIKELY (dir == NULL))
+    return crun_make_error (err, 0, "cannot get state directory");
+
+  ret = append_paths (&src_path, err, dir, "seccomp.bpf", NULL);
+  if (UNLIKELY (ret < 0))
+    return ret;
+
+  ret = append_paths (&seccomp_cache_dir, err, dir, "..", ".seccomp-cache", NULL);
+  if (UNLIKELY (ret < 0))
+    return ret;
+
+  ret = stat (seccomp_cache_dir, &st);
+  if (ret == 0 && st.st_size > 500)
+    {
+      cleanup_dir DIR *d = NULL;
+      size_t cleaned = 0;
+      struct dirent *de;
+
+      d = opendir (seccomp_cache_dir);
+      if (d == NULL)
+        return crun_make_error (err, errno, "cannot open seccomp cache directory");
+      while ((de = readdir (d)))
+        {
+          cleanup_free char *seccomp_cache = NULL;
+
+          if (de->d_name[0] == '.')
+            continue;
+
+          ret = append_paths (&seccomp_cache, err, seccomp_cache_dir, de->d_name, NULL);
+          if (UNLIKELY (ret < 0))
+            return ret;
+
+          ret = unlink (seccomp_cache);
```

Comment:
I wonder if this should perhaps look at mtimes and remove the oldest ones instead? I feel like this could lead to ping-ponging the last added items.

Reply:
pushed a new version